internal/cmd/controller/agentmanagement/controllers/cluster/import.go

```diff
@@ -97,7 +97,10 @@ func (i *importHandler) onConfig(config *config.Config) error {
 
 		hasConfigChanged := config.APIServerURL != cluster.Status.APIServerURL ||
 			hashStatusField(config.APIServerCA) != cluster.Status.APIServerCAHash ||
-			config.AgentTLSMode != cluster.Status.AgentTLSMode
+			config.AgentTLSMode != cluster.Status.AgentTLSMode ||
+			(config.GarbageCollectionInterval.Duration != 0 && cluster.Status.GarbageCollectionInterval == nil) ||
+			(cluster.Status.GarbageCollectionInterval != nil &&
+				config.GarbageCollectionInterval.Duration != cluster.Status.GarbageCollectionInterval.Duration)
```

Comment:
nit: maybe create a helper function that do those checks about `GarbageCollectionInterval`.
I found it hard to read as it is a pretty large expression, the other checks are pretty simple in comparison.

Reply:
Good point, thanks for raising this. I've moved this to a separate function.